k2/csrc/rnnt_decode.h

```diff
@@ -108,10 +109,21 @@ struct ArcInfo {
 };
 
 struct RnntDecodingStream {
-  // `graph` is a pointer to the FSA (decoding graph) that we are decoding this
-  // stream with.  Different streams might have different graphs.  This must
-  // be an Fsa, not FsaVec (i.e. 2 axes).
-  std::shared_ptr<Fsa> graph;
+  // construct a RnntDecodingStream from the decoding graph
+  explicit RnntDecodingStream(const Fsa &fsa) : graph(fsa) {
+    num_graph_states = graph.shape.Dim0();
+    ContextPtr &c = graph.Context();
+    // initialize to start state
+    states = Ragged<int64_t>(RegularRaggedShape(c, 1, 1),
+                             Array1<int64_t>(c, std::vector<int64_t>{0}));
+    scores = Ragged<double>(states.shape,
+                            Array1<double>(c, std::vector<double>{0.0}));
+  }
+
+  // `graph` is a reference to the FSA (decoding graph) that we are decoding
+  // this stream with.  Different streams might have different graphs.
+  // This must be an Fsa, not FsaVec (i.e. 2 axes).
+  const Fsa &graph;
```

Comment:
`Fsa` contains some shared pointers and is cheap to copy, I think.

It is not clear here who is repsonbile for managing the lifetime of `graph` if it is a reference.